web/src/features/charts/bar-breakdown/utils.ts

```diff
@@ -131,29 +115,30 @@ export const getDataBlockPositions = (
 
 export interface ExchangeDataType {
   exchange: number;
-  mode: ZoneKey; // TODO: Weird that this is called "mode"
-  gCo2eqPerkWh: number;
-  tCo2eqPerMin: number;
+  exchangeCapacityRange: number[];
+  mode: ZoneKey; // TODO: This should not be called "mode" as it's a zonekey
+  gCo2eqPerkWh: Maybe<number>;
+  tCo2eqPerMin: Maybe<number>;
 }
 export const getExchangeData = (
   data: ZoneDetail,
-  exchangeKeys: string[],
-  electricityMixMode: Mode
+  exchangeKeys: ZoneKey[],
+  electricityMixMode: MixMode
 ): ExchangeDataType[] =>
-  exchangeKeys.map((mode) => {
+  exchangeKeys.map((key) => {
     // Power in MW
-    const exchange = (data.exchange || {})[mode];
-    const exchangeCapacityRange = (data.exchangeCapacities || {})[mode];
+    const exchange = (data.exchange || {})[key];
+    const exchangeCapacityRange = (data.exchangeCapacities || {})[key];
```

Comment:
Shouldn't this use optional chaining instead?
```suggestion
    const exchange = data.exchange?.[key];
    const exchangeCapacityRange = data.exchangeCapacities?.[key];
```